ArcGISRuntimeSDKQt_QMLSamples/CloudAndPortal/IntegratedWindowsAuthentication/IntegratedWindowsAuthentication.qml

```diff
@@ -0,0 +1,241 @@
+            top: parent.top
+        }
+        width: childrenRect.width
+        height: childrenRect.height
+        color: "#000000"
+        opacity: .70
+        radius: 5
+
+        ColumnLayout {
+            TextField {
+                id: securePortalUrl
+                Layout.fillWidth: true
+                Layout.margins: 2
+                placeholderText: qsTr("Enter portal url secured by IWA")
+                background: Rectangle {
+                    color: "white"
+                }
+            }
+
+            Row {
+                Layout.alignment: Qt.AlignHCenter | Qt.AlignVCenter
+                CheckBox {
+                    id: forceLoginBox
+                    checked: false
+                }
+                Text {
+                    text: qsTr("Force login")
+                    height: forceLoginBox.height
+                    verticalAlignment: Text.AlignVCenter
+                    color: "#ffffff"
+                }
+            }
+
+            Row {
+                Layout.alignment: Qt.AlignHCenter | Qt.AlignVCenter
+                Layout.margins: 2
+                spacing: 3
+
+                Button {
+                    text: qsTr("Search Public")
+                    onClicked: {
+                        searchPortal(arcgis_url, false);
+                    }
+                }
+                Button {
+                    text: qsTr("Search Secure")
+                    onClicked: {
+                        if (securePortalUrl.text) {
+                            searchPortal(securePortalUrl.text, forceLoginBox.checked);
+                        } else {
+                            webMapMsg.text = "Portal URL is empty. Please enter a portal URL"
+                            webMapMsg.visible = true;
+                            return;
+                        }
+                    }
+                }
+            }
+
+            ComboBox {
+                id:webmapsList
+                Layout.margins: 2
+                Layout.fillWidth: true
+                model: null
+            }
+
+            Button {
+                text: qsTr("Load Web Map")
+                Layout.fillWidth: true
+                Layout.margins: 2
+                onClicked: {
+                    if (webmapsList.currentIndex >= 0) {
+                        indicator.running = true;
+                        loadSelectedWebmap(portalItemListModel.get(webmapsList.currentIndex));
+                    }
+                }
+            }
+        }
+    }
+
+    BusyIndicator {
+        id: indicator
+        anchors.centerIn: parent
+        running: false
+    }
+
+    PortalQueryParametersForItems {
+        id: webmapQuery
+        types: [ Enums.PortalItemTypeWebMap ]
+    }
+
+    function searchPortal (portalUrl, forceLogin) {
+
+        var portal;
+        if (forceLogin) {
+            portal = ArcGISRuntimeEnvironment.createObject("Portal", {url: portalUrl, loginRequired: forceLogin});
```

Comment:
same comment as in C++ - you shouldn't need to create a new instance every time. The Portal object isn't really meant to be transient but rather, you would normally have 1 portal instance in your app that you've authenticated with and then you issue your commands on that one instance (query, add, remove, etc). 

I suggest you create the instance once only. Since this is QML, I'd recommend creating it declaratively and connecting to signals there instead of in JS code as it is a little easier to follow and to handle JS scoping/GC issues